common/app/common/ApplicationMode.scala

```diff
@@ -0,0 +1,19 @@
+package common
+
+import play.api.{Application, GlobalSettings}
+import play.api.Mode._
+
+object ApplicationMode {
+  private var theMode = Prod
```

Comment:
I agree on not using an option with a defensive default value. It keeps the logic simple and ensure if we ever run into a problem, it'll be either when testing or coding locally (but not in prod)

You're suggesting we keep a global reference to a play.api.Application instead of the Mode? That would simplify the migration but I believe play 2.5 is getting away from that object that rules them all